File: Drivo/src/app/services/resources.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { firstValueFrom } from 'rxjs';
import { ResourceEntity } from 'src/entities/resource.entity';
import { environment } from 'src/environments/environment';
import { ActionResponse } from 'src/responses/action.response';

@Injectable({
  providedIn: 'root'
})
export class ResourcesService {

  constructor(private httpClient: HttpClient) { }

  public async addResourceAsync(resource: ResourceEntity): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.post<ActionResponse>(`${environment.apiUri}/Resources`, resource));
  }
  
  public async updateResourceAsync(resource: ResourceEntity): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.put<ActionResponse>(`${environment.apiUri}/Resources`, resource));
  }

  public async removeResourceAsync(resource: ResourceEntity): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.delete<ActionResponse>(`${environment.apiUri}/Resources/${resource.id}`));
  }

}

File: Drivo.WebAPI/Services/ResourcesService.cs
using Drivo.Entities;
using Drivo.Responses;
using Drivo.WebAPI.Repositories;

namespace Drivo.WebAPI.Services;

public class ResourcesService
{
    public ResourcesService(ResourcesRepository resourcesRepository)
    {
        ResourcesRepository = resourcesRepository;
    }

    private ResourcesRepository ResourcesRepository { get; }

    public async Task<ResourceEntity> GetResourceByIdAsync(int resourceId)
    {
        return await ResourcesRepository.GetResourceByIdAsync(resourceId);
    }

    public async Task<ActionResponse> AddResourceAsync(ResourceEntity resource)
    {
        return await ResourcesRepository.UpdateResourceAsync(resource);
    }

    public async Task<ActionResponse> UpdateResourceAsync(ResourceEntity resource)
    {
        return await ResourcesRepository.UpdateResourceAsync(resource);
    }

    public async Task<ActionResponse> RemoveResourceAsync(int resourceId)
    {
        return await ResourcesRepository.RemoveResourceAsync(await GetResourceByIdAsync(resourceId));
    }
}
